### PR TITLE
Run lxc commands as non root by default

### PR DESCRIPTION
**Pull request by lucasmoura**
In ubuntu-advantage-tools, we require some LXD commands to be run as `non-root`. This PR changes the default behavior of LXD to run exec commands as non-root.

This will impact some `cloud-init` tests, but we already have a [PR](https://github.com/canonical/cloud-init/pull/664) to handle that

### CODE REVIEW

**Comment by TheRealFalcon** [comment]
Actually...lxd is the exception for us as all other clouds already execute non-root. This may actually simplify things for us :slightly_smiling_face:

**Review by TheRealFalcon — APPROVED** [review]
Travis is failing as this will break cloud-init tests, but once https://github.com/canonical/cloud-init/pull/664 lands it should be a one-line fix. Since we've pinned our pycloudlib version, I'm not concerned these failures.

**Review by blackboxsw — APPROVED** [review]
Once https://github.com/canonical/cloud-init/pull/694 lands, this looks good to land

**Review by OddBloke — APPROVED** [review]
The conclusion we came to in https://github.com/canonical/cloud-init/pull/664 was that `pycloudlib` should consistently execute commands as an unprivileged user by default. This aligns LXD with the other clouds in pycloudlib which use SSH.

**Review by OddBloke — CHANGES_REQUESTED** [review]
Per https://github.com/canonical/cloud-init/pull/694#discussion_r531159622, running commands as UID 1000 is _not_ the same as running commands as `self.username`. I believe this will need modification to look up the UID of `self.username` before running `lxc exec` for the first time.

**Comment by lucasmoura** [comment]
@OddBloke I have updated the code to get the `user id`  before running any `lxc exec` commands.

The solution I have is to retry the `id -u` command until the user was created. With that solution, I think we can drop from LXD the call for [_wait_for_execute](https://github.com/canonical/pycloudlib/blob/master/pycloudlib/instance.py#L369), since the `user_id` property is kind of covering that already

File: pycloudlib/lxd/instance.py [hunk 1]
```diff
@@ -23,6 +23,7 @@ def __init__(self, name, key_pair=None):
         super().__init__(key_pair=key_pair)
 
         self._name = name
+        self._user_id = None
```
Comment (OddBloke) [inline pycloudlib/lxd/instance.py:26]:
```suggestion
```

**Review by OddBloke — APPROVED** [review]
Thanks!

**Pull request merged**